### PR TITLE
[Ray] Loads the subtask inputs from meta

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?

This PR makes the Ray executor able to load the subtask inputs from meta, so Ray backend can run the task which inputs are the executed tasks.

## Related issue number

https://github.com/mars-project/mars/issues/2893

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/deploy/oscar/tests/test_local.py [hunk 1]
```diff
@@ -421,23 +424,26 @@ def test_sync_execute():
...
+                result = df.sum(axis=1).execute().fetch()
+                expected = pd.read_csv(file_path).sum(axis=1)
+                pd.testing.assert_series_equal(result, expected)
+
+                df = md.read_csv(file_path, chunk_bytes=os.stat(file_path).st_size / 5)
+                result = df.head(10).execute().fetch()
+                expected = pd.read_csv(file_path).head(10)
+                pd.testing.assert_frame_equal(result, expected)
+        else:
+            print("Incremental index is not supported.")
```
Comment (qinxuye) [inline mars/deploy/oscar/tests/test_local.py:446]: Is this print intended or just for test?

Comment (fyrestone) [reply mars/deploy/oscar/tests/test_local.py:446]: I think it's better to print this if the incremental index is not supported / tested. I am working on the Ray backend to support the incremental index feature as soon as possible. There is a TODO comment in the Ray test case.

Comment (qinxuye) [reply mars/deploy/oscar/tests/test_local.py:446]: How about add a TODO instead of printing sth?

Comment (fyrestone) [reply mars/deploy/oscar/tests/test_local.py:446]: There is  already a TODO comment in the Ray test case: https://github.com/mars-project/mars/pull/2976/files#diff-c1dddc59766fc396dab0a8b1c8f3ffe0a9256c72a07feb11d76b0239fff5a437R106

I will remove this print as you have concerns.

Comment (fyrestone) [reply mars/deploy/oscar/tests/test_local.py]: The print has been removed.

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Pull request merged**